spec/pacio_test_kit/create_test_spec.rb

```diff
@@ -0,0 +1,118 @@
+RSpec.describe PacioTestKit::CreateTest do
+  let(:runnable) do
+    Class.new(PacioTestKit::CreateTest) do
+      config(
+        options: {
+          resource_type: 'Observation',
+          profile: 'PFESingleObservation'
+        }
+      )
+
+      input :url
+
+      fhir_client do
+        url :url
+      end
+    end
+  end
+  let(:session_data_repo) { Inferno::Repositories::SessionData.new }
+  let(:results_repo) { Inferno::Repositories::Results.new }
+  let(:test_session) { repo_create(:test_session, test_suite_id: 'pacio_pfe_server') }
+  let(:url) { 'https://example/r4' }
+  let(:resource_type) { 'Observation' }
+  let(:resource_id) { '123' }
+  let(:observation) do
+    JSON.parse(
+      File.read(File.join(__dir__, '..', 'fixtures', 'pfe_single_observation.json'))
+    )
+  end
+  # let(:resource_header) { Inferno::Entities::Header }
+  # let(:resource_header_params) {resource_header.id = 'example_id', resource_header.vid = 'example_vid', }
+
+  def run(runnable, inputs = {})
+    test_run_params = { test_session_id: test_session.id }.merge(runnable.reference_hash)
+    test_run = Inferno::Repositories::TestRuns.new.create(test_run_params)
+    inputs.each do |name, value|
+      session_data_repo.save(
+        test_session_id: test_session.id,
+        name:,
+        value:,
+        type: runnable.config.input_type(name)
+      )
+    end
+    Inferno::TestRunner.new(test_session:, test_run:).run(runnable)
+  end
+
+  def entity_result_message
+    results_repo.current_results_for_test_session_and_runnables(test_session.id, [runnable])
+      .first
+      .messages
+      .first
+  end
+
```

Comment:
Look add the [webmock  gem doc](https://github.com/bblimke/webmock#stubbing-with-custom-response) to see how to do stubbing with custom response, including headers.

For the checks, you should have the following:
- it passes when resource is successfully created with the correct metadata and location header is returned
- it fails when the submitted resource is not a valid json
- it fails when the submitted resource does not include the `resourceType` field
- it fails when the submitted resource type is not the expected resource type
- it fails if the response status is not 201
- it fails if the response body is not a valid json
- it fails if the response resource type is not the expected type
- it fails if the response resource id is missing
- it fails if the response resource id is the same as the submitted resource id
- it fails if the response resource meta.versionId is missing
- it fails if the response resource meta.versionId is the same as the submitted resource meta.versionId
- it fails if the response resource meta.lastUpdated is missing
- it fails if the response resource  meta.lastUpdated is the same as the submitted resource  meta.lastUpdated 
- it fails if location header is not returned
- it fails if the location header value does not include the correct resource type
- it fails if the location header value does not include the correct resource id